src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,312 @@
+            Initialize();
+        }
+
+        public void BatchBegin()
+        {
+        }
+
+        public void BatchCommit()
+        {
+        }
+
+        public new void SetContent(EvasObject content)
+        {
+            _contentBox.PackEnd(content);
+
+            if (content is NavigationView nv)
+            {
+                nv.Dragged += (s, e) =>
+                {
+                    if (e.State == DraggedState.EdgeTop)
+                    {
+                        Close();
+                    }
+                };
+            }
+        }
+
+        public async void Open(uint length = 300)
+        {
+            var toMove = _outterBox.Geometry;
+            toMove.Y = 0;
+
+            await RunMoveAnimation(_outterBox, toMove, length);
+
+            if (!_isOpen)
+            {
+                _isOpen = true;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async void Close(uint length = 300)
+        {
+            var toMove = _outterBox.Geometry;
+            toMove.Y = Geometry.Height - MinimumDrawerHeight;
+
+            await RunMoveAnimation(_outterBox, toMove, length);
+
+            if (_isOpen)
+            {
+                _isOpen = false;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async Task<bool> ShowAsync(EWidget target, Easing easing = null, uint length = 300, CancellationToken cancelltaionToken = default(CancellationToken))
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            await Task.Delay(1000);
+
+            if (cancelltaionToken.IsCancellationRequested)
+            {
+                cancelltaionToken.ThrowIfCancellationRequested();
+            }
+
+            _outterBox.Show();
+            var opacity = _outterBox.Opacity;
+
+            new Animation((progress) =>
+            {
+                _outterBox.Opacity = opacity + (int)((255 - opacity) * progress);
+
+            }).Commit(this, "FadeIn", length: length, finished: (p, e) =>
+            {
+                _outterBox.Opacity = 255;
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        public async Task<bool> HideAsync(EvasObject target, Easing easing = null, uint length = 300)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var opacity = _outterBox.Opacity;
+            new Animation((progress) =>
+            {
+                _outterBox.Opacity = opacity - (int)(progress * opacity);
+
+            }).Commit(this, "FadeOut", length: length, finished: (p, e) =>
+            {
+                _outterBox.Opacity = 0;
+                _outterBox.Hide();
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            _gestureBox = new Box(parent);
```

Comment:
`_gestureBox` naming is not good, it should be role of main layout

Reply:
Fixed. The name is changed to _mainLayout